tests/compile-fail/lifetimes.rs

```diff
@@ -0,0 +1,70 @@
+#![feature(plugin)]
+#![plugin(clippy)]
+
+#![deny(needless_lifetimes)]
+
+fn distinct_lifetimes<'a, 'b>(_x: &'a u8, _y: &'b u8, _z: u8) { } //~ERROR
```

Comment:
It is good custom to copy the start of the error message after the `//~ERROR` (if you dislike the width, you can also write `//~^ERROR` in the next line. This also allows to declare multiple errors in one line, just increment the number of ^). I recently failed to follow this good custom and got a successful compile-fail test when the lint in fact didn't work.